File: src/main/java/Sigmath/engineering/mechanical.java
package sigmath;



class Mechanical
{
    public static double calculateTorque(double force, double radius)
    {
        return force * radius;
    }

    public static double calculatePower(double torque, double angularVelocity)
    {
        return torque * angularVelocity;
    }

    public static double calculateVelocity(double distance, double time)
    {
        return distance / time;
    }

    public static double calculateAcceleration(double initialVelocity, double finalVelocity, double time)
    {
        return (finalVelocity - initialVelocity) / time;
    }

    public static double calculateForce(double mass, double acceleration)
    {
        return mass * acceleration;
    }
}

File: cpp_build/source/engineering/mechanical.cpp
#include "mechanical.h"

namespace Sigmath {

RTTI_INFO_IMPL_HASH(473036584u, ::Sigmath::Mechanical, ThisTypeBaseTypesInfo);

double Mechanical::CalculateTorque(double force, double radius)
{
    return force * radius;
}

double Mechanical::CalculatePower(double torque, double angularVelocity)
{
    return torque * angularVelocity;
}

double Mechanical::CalculateVelocity(double distance, double time)
{
    return distance / time;
}

double Mechanical::CalculateAcceleration(double initialVelocity, double finalVelocity, double time)
{
    return (finalVelocity - initialVelocity) / time;
}

double Mechanical::CalculateForce(double mass, double acceleration)
{
    return mass * acceleration;
}

} // namespace Sigmath
